### ISSUE TITLE
sudoers.d template has incorrect formatting for command with no password

### ISSUE THREAD

**Issue by efalkner** [chef-cookbooks/sudo#60]
Hello,

I've tried using the sudo cookbook to create a sudoers.d file for a user. I'd like to be able to run one command as root with no password.
The template creates a new sudoers.d file for me, but the user was still being prompted for a password. When I dug a bit further, it turned out that the template's format was creating this line:  
     user ALL=(ALL) :NOPASSWD /command/foo
where it should be writing this line:
     user ALL= :NOPASSWD /command/foo

I've also tried adding a local template and running the sudo cookbook with this config:

```
sudo 'user' do
  template 'newsudoer.erb'
  variables( :cmds => ['/etc/init.d/service'] )
end
```

This, however, did not pick up the commands array as variables, and failed my kitchen run with the error:
     NoMethodError: undefined method `each' for nil:NilClass

The current stack I'm working with is:
Chef Omibus 11.16.4
Chef Zero
and platform Ubuntu 14.04

As of now, I am locked in to my current chef version.
Any help / suggestions would be appreciated! Thanks!

**Comment by kungfujoe** [comment]
Pretty sure you did it wrong. Note that I'm not involved in this project, I just happened to stumble in, and I know how to do this. I had to add two LWRPs to make this work. I wanted to make user 'foo' a sudoer that requires a password for all commands except chef-client. This is how I did it:

```
sudo 'foo-as-sudoer' do
  user 'foo'
  nopasswd false
end

sudo 'foo-run-chef-client-nopasswd' do
  user 'foo'
  commands ['/usr/bin/chef-client']
  runas 'root'
  nopasswd true
end
```

I've verified this code on a fresh machine within the last half hour, so I hope this works out for you!
